### PR TITLE
Aggregate supports group by date fields

### PR DESCRIPTION
**Pull request by davimacedo**
This PR aims to affirm that Parse Server currently DOES support group by date fields.

I started working on this because I've just setup a brand new environment for contributing to Parse Server and the test below keeps failing to me when I run coverage npm script on master:
https://github.com/parse-community/parse-server/blob/master/spec/ParseQuery.Aggregate.spec.js#L381

I found out that the group by date fields actually works for normal cases and only does not work when you have dirty data, meaning a non date value in a field that is expected to be a date. I was only able to simulate this problem by doing:
1) I dropped the test database
2) I used the api to save a data in the TestObject class with value '' in the field dateField
3) I run the tests

After this, I've just dropped the database again and run the tests. Then the group by was working again.

Probably this test is passing in Travis and maintainers' environments because they have legacy data in their databases. According to my tests, if any of them drop the current test database and run the tests again, this test will fail.

In summary, what I did in this PR:
- Changed the test to affirm that group by field dates actually works;
- Changed the date field name to dateField2 to avoid problem in environments with legacy dirty data;
- Removed the error handler below, since the error is related to dirty data and not bad query; it will make the original mongodb error to be thrown and I think it will be easier for the developer to find out what is happening since it is possible to find some information about this error in internet;
https://github.com/parse-community/parse-server/blob/master/src/Adapters/Storage/Mongo/MongoStorageAdapter.js#L768

@dplewis do you mind to take a look on this and check if you agree?

### CODE REVIEW

**Comment by dplewis** [comment]
Did you figure something out?

**Comment by davimacedo** [comment]
The tests are passing in my environment but didn't pass in Travis. I'm trying to understand why.

File: spec/ParseQuery.Aggregate.spec.js [hunk 1]
```diff
@@ -401,11 +401,13 @@ describe('Parse.Query Aggregate testing', () => {
           const query = new Parse.Query(TestObject);
           return query.aggregate(pipeline);
         })
-        .then(done.fail)
-        .catch(error => {
-          expect(error.code).toEqual(Parse.Error.INVALID_QUERY);
+        .then(results => {
+          const counts = results.map(result => result.count);
+          expect(counts.length).toBe(3);
+          expect(counts.sort()).toEqual([1, 2, 4]);
```
Comment (dplewis) [inline spec/ParseQuery.Aggregate.spec.js]: If group by date works, wouldn't the results be obj1 + obj3 is one group and obj2 is another.

count would be [1, 2] what does the 4 come from?

**Comment by dplewis** [comment]
Which version of mongo are you running locally?

File: spec/ParseQuery.Aggregate.spec.js (hunk 1 above)
Comment (davimacedo) [reply spec/ParseQuery.Aggregate.spec.js]: 4 come from the default data that is added before each test. It returns [ objectId: undefined, count: 4]

**Comment by davimacedo** [comment]
Antonios-MacBook-Pro-2:parse-server davimacedo$ mongod --version
db version v4.0.9
git version: fc525e2d9b0e4bceff5c2201457e564362909765
allocator: system
modules: none
build environment:
    distarch: x86_64
    target_arch: x86_64

**Comment by davimacedo** [comment]
I think in this new version I'm running mongo works for undefined values, but it doesn't for version 4.0.4. I just added the match state to ensure that the undefined dates (the ones added before each test) are not processed. Let's see if it will pass in Travis now.

File: spec/ParseQuery.Aggregate.spec.js (hunk 1 above)
Comment (davimacedo) [reply spec/ParseQuery.Aggregate.spec.js]: This data here: https://github.com/parse-community/parse-server/blob/master/spec/ParseQuery.Aggregate.spec.js#L21

**Comment by davimacedo** [comment]
It passed. Conclusion:
- for mongo 4.0.9 it works except in the case that you have values that are not date, but works even if you have values that are undefined
- for mongo <=4.0.4 it works except in the case that you have values that are not date or are undefined

In both scenario you can add a match stage to filter out the data before the groupBy.

File: spec/ParseQuery.Aggregate.spec.js [hunk 2]
```diff
@@ -379,18 +379,23 @@ describe('Parse.Query Aggregate testing', () => {
   });
 
   it_exclude_dbs(['postgres'])(
```
Comment (dplewis) [inline spec/ParseQuery.Aggregate.spec.js]: This should work on postgres right?

Comment (davimacedo) [reply spec/ParseQuery.Aggregate.spec.js]: I am not sure. I will check.

Comment (davimacedo) [reply spec/ParseQuery.Aggregate.spec.js]: The group by date field works, but the test fails because the $exists is not implemented for match stage on postgres. I included an additional test only for postgres.

**Comment by dplewis** [comment]
@davimacedo Thanks!
